code/datums/datumvars.dm

```diff
@@ -386,7 +386,10 @@ body
 				html += "<ul>"
 				var/index = 1
 				for(var/entry in L)
-					html += debug_variable(index, L[index], level + 1)
+					if(istext(entry))
+						html += debug_variable(entry, L[index], level + 1)

```

Comment:
This wouldn't work. You need to do `L[entry]` not `L[index]` Also, not all associative lists use text as keys.